ckan/lib/search/query.py

```diff
@@ -89,6 +89,27 @@ def convert_legacy_parameters_to_solr(
     return solr_params
 
 
+def _get_local_query_parser(q: str) -> str:
+    """
+    Given a Solr `q` parameter, extract any custom query parsers used in the
+    local parameters, .e.g. q={!child ...}...
+    """
+    qp_type = ""
+    q = q.strip()
+    if not q.startswith("{!"):
+        return qp_type
+
+    parts = q.lstrip("{!").rstrip("}").split(" ")
+    if "=" not in parts[0]:
+        # Most common form of defining the query parser type e.g. {!knn ...}
+        qp_type = parts[0]
+    else:
+        # Alternative syntax e.g. {!type=knn ...}
+        type_part = [p for p in parts if p.startswith("type=")]
+        if type_part:
+            qp_type = type_part[0].split("=")[1]
```

Comment:
what does multiple `type=` parameters do? Is it safe to assume solr uses the first?